.github/workflows/tests-windows.yaml

```diff
@@ -0,0 +1,35 @@
+name: Run cli and unit tests on Windows
+
+on:
+    push:
+        branches:
+            - master
+            - staging/**
+    pull_request:
+
+concurrency:
+    group: tests-${{ github.event.pull_request.number || github.ref }}
+    cancel-in-progress: true
+
+jobs:
+    tests:
+        runs-on: windows-latest
+        strategy:
+            matrix:
+                node-version: [18.x, 20.x]
```

Comment:
I think it's possible to use the existing `tests.yaml` and add the os in the matrix 

example https://github.com/vitest-dev/vitest/blob/main/.github/workflows/ci.yml